astropy/cosmology/io/row.py

```diff
@@ -57,24 +58,33 @@ def from_row(row, *, move_to_meta=False, cosmology=None):
 
         >>> cosmo = Cosmology.from_format(cr, format="astropy.row")
         >>> cosmo
-        FlatLambdaCDM(name="Planck18", H0=67.7 km / (Mpc s), Om0=0.31,
-                      Tcmb0=2.725 K, Neff=3.05, m_nu=[0. 0. 0.06] eV, Ob0=0.049)
+        FlatLambdaCDM(name="Planck18", H0=67.66 km / (Mpc s), Om0=0.30966,
+                      Tcmb0=2.7255 K, Neff=3.046, m_nu=[0. 0. 0.06] eV, Ob0=0.04897)
     """
     # special values
     name = row['name'] if 'name' in row.columns else None  # get name from column
-    meta = copy.deepcopy(row.meta)
+
+    meta = defaultdict(dict, copy.deepcopy(row.meta))
+    # Now need to add the Columnar metadata. This is only available on the
+    # parent table. If Row is ever separated from Table, this should be moved
+    # to ``to_table``.
+    for col in row._table.itercols():
+        # Only add metadata if exists and not empty,
+        # e.g. Table has Column with meta, QTable doesn't have Column.
+        if getattr(col, "meta", None):
```

Comment:
the way to get `meta` independent of the type of column is to use `col.info.meta` (which `Quantity` has too).

Reply:
Great! did not know about this. Very convenient.